src/api/UploadAPI.ts

```diff
@@ -1,129 +1,136 @@
-import info from './info'
 import UploadClient from '../UploadClient'
-import fromUrlStatus, { FromUrlStatusResponse } from './fromUrlStatus'
-import base, { BaseResponse } from './base'
-import {
-  FileInfoInterface,
-  GroupId,
-  GroupInfoInterface,
-  Token,
-  UploadAPIInterface,
-  Uuid
-} from './types'
-import { RequestOptionsInterface, RequestResponse } from './request/types'
-import fromUrl, { FromUrlResponse, Url } from './fromUrl'
+import request from './request/request.node'
+import base from './base'
+import info from './info'
+import fromUrl from './fromUrl'
+import fromUrlStatus from './fromUrlStatus'
+import group, { GroupOptions } from './group'
 import groupInfo from './groupInfo'
-import { FileData, SettingsInterface } from '../types'
-import { prepareOptions } from './request/prepareOptions'
-import group from './group'
-import request from './request/request'
-import { MultipartPart, MultipartStartResponse } from './multipart/types'
-import multipartComplete from './multipart/multipartComplete'
-import multipartStart from './multipart/multipartStart'
-import multipartUpload from './multipart/multipartUpload'
-import {
-  BaseThenableInterface,
-  CancelableThenableInterface
-} from '../thenable/types'
-import { BaseHooksInterface, CancelHookInterface } from '../lifecycle/types'
-
-class UploadAPI implements UploadAPIInterface {
-  readonly client: UploadClient
+
+/* Types */
+import { BaseOptions, FileData, BaseResponse } from './base'
+import { RequestOptions, RequestResponse } from './request/request.node'
+import { FileInfo, GroupId, GroupInfo, Token, Url, Uuid } from './types'
+import { InfoOptions } from './info'
+import { FromUrlOptions, FromUrlResponse } from './fromUrl'
+import { FromUrlStatusOptions, FromUrlStatusResponse } from './fromUrlStatus'
+import { SettingsInterface } from '../types'
+
+/**
+ * Populate options with settings.
+ * @param {<T>} options
+ * @param {SettingsInterface} settings
+ */
+const populateOptionsWithSettings = <T>(
+  options: T,
+  settings: SettingsInterface
+): T => ({
+  ...settings,
+  ...options
+})
+
+class UploadAPI {
+  private client: UploadClient
 
   constructor(client: UploadClient) {
     this.client = client
   }
 
-  private getExtendedSettings = (settings): SettingsInterface => {
-    return {
-      ...this.client.getSettings(),
-      ...settings
-    }
+  request(options: RequestOptions): Promise<RequestResponse> {
+    return request(options)
   }
```

Comment:
I don't think that we need provide access to `request` here, because `request` builded for send only uploadcare specific requests. Please remove this method 🙌🏻

Reply:
Ok, I agree with you.